Sources/Engine/Sources/LayoutThatFits.swift

```diff
@@ -105,9 +105,12 @@ public struct LayoutThatFits: Layout {
     ) {
         var stop = false
         var index = 0
-        while !stop {
-            stop = index == layouts.count - 1
+        while !stop && index < layouts.count {
+            let isLast = index == layouts.count - 1
             accessor(&stop, &cache.caches[index], layouts[index])
+            if isLast {
```

Comment:
This does not improve efficiency. Stop is already set to true when visiting the last layout. 

This change makes it so that it is not true, which will cause regressions since now the visitor won't use the layout if it doesn't fit. But it's intentional to use the last layout if no other layout fits.